src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Parses the user input and executes the corresponding commands.
+ */
+public class Parser {
+    /**
+     * Returns Command object based on user's input.
+     * @param fullCommand user's input.
+     * @return specific command.
+     * @throws DukeException If command does not exist.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] commandSegments = fullCommand.split(" ", 2);
+        String mainCommand = commandSegments[0].toLowerCase().trim();
+
+        switch (mainCommand) {
+        case "list":
+            return new ListCommand();
+        case "bye":
+            return new ByeCommand();
+        case "todo":
+            return new TodoCommand(commandSegments[1].trim());
+        case "deadline":
+            String[] deadlineSegments = commandSegments[1].split("/by", 2);
+            return new DeadlineCommand(deadlineSegments[0], LocalDate.parse(deadlineSegments[1].trim()));
+        case "event":
+            String[] eventSegments = commandSegments[1].split("/at", 2);
+            return new EventCommand(eventSegments[0], eventSegments[1]);
```

Comment:
The deadline with one date specified and the event with two dates specified are treated differently here. Have you considered splitting the two dates for Event or not parsing the date before instantiating a `DeadlineCommand`?